ReactNativeClient/lib/joplin-database.js

```diff
@@ -313,7 +313,11 @@ class JoplinDatabase extends Database {
 
 		// currentVersionIndex < 0 if for the case where an old version of Joplin used with a newer
 		// version of the database, so that migration is not run in this case.
-		if (currentVersionIndex < 0) throw new Error('Unknown profile version. Most likely this is an old version of Joplin, while the profile was created by a newer version. Please upgrade Joplin at https://joplinapp.org and try again.');
+		if (currentVersionIndex < 0) {
+			this.logger().error(`Cannot convert database from version ${fromVersion} to ${existingDatabaseVersions[existingDatabaseVersions.length-1]}`);
+			const p = require('../package.json');
+			throw new Error(`Unknown profile version. Most likely this is an old version of Joplin, while the profile was created by a newer version. Please upgrade Joplin at https://joplinapp.org and try again.\n${p.name} version: ${p.version}\nProfile version: ${fromVersion}\nExpected version: ${existingDatabaseVersions[existingDatabaseVersions.length-1]}`);
```

Comment:
Please use packageInfo here instead. As it is used in `app.js`.

Reply:
That works for ElectronClient build, but not for CliClient.  It looks like packageInfo.js only exists in the ElectronClient build.  Is that right?

Reply:
Hmm, you might be right. I thought it was made available to all platforms, but I mixed it up with something else.